### PR TITLE
Issue horizontal bar

### PR DESCRIPTION
**Pull request by mengweieric**
### Description
- horizontal backward compatibility
- line fixes
- thresholds adjustments

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-observability/public/components/explorer/visualizations/config_panel/config_editor/config_controls/config_thresholds.tsx [hunk 1]
```diff
@@ -30,7 +31,7 @@ export const ConfigThresholds = ({
   let addButtonText = '+ Add threadshold';
   const getThresholdUnit = () => {
     return {
-      thid: uniqueId('thr'),
+      thid: htmlIdGenerator('thr'),
```
Comment (eugenesk24) [inline dashboards-observability/public/components/explorer/visualizations/config_panel/config_editor/config_controls/config_thresholds.tsx:34]: Just curious: Why the change??

File: dashboards-observability/public/components/explorer/visualizations/config_panel/config_editor/config_controls/config_thresholds.tsx [hunk 2]
```diff
@@ -102,43 +103,53 @@ export const ConfigThresholds = ({
               <EuiFormRow fullWidth label="">
                 <EuiFlexGroup alignItems="center" gutterSize="xs">
                   <EuiFlexItem grow={3}>
-                    <EuiColorPicker
-                      fullWidth
-                      onChange={handleThresholdChange(thr.thid, 'color')}
-                      color={thr.color}
-                    />
+                    <EuiFormRow helpText="color">
```
Comment (eugenesk24) [inline dashboards-observability/public/components/explorer/visualizations/config_panel/config_editor/config_controls/config_thresholds.tsx:106]: Is wrapping them in `EuiFormRow` just to add the helpText or does it change the look as well??

**Review by eugenesk24 — APPROVED** [review]
:shipit:

**Review by ps48 — DISMISSED** [review]

File: dashboards-observability/public/components/explorer/visualizations/config_panel/config_panel.tsx [hunk 3]
```diff
@@ -173,6 +173,10 @@ export const ConfigPanel = ({ visualizations, setCurVisId }: any) => {
     [memorizedVisualizationTypes]
   );
 
+  const vizTypeList = useMemo(() => {
```
[the remark this answers is not in the document]
Comment (mengweieric) [reply dashboards-observability/public/components/explorer/visualizations/config_panel/config_panel.tsx:176]: this is a quick change for the existing code base to add horizontal type support for 1.2 saved visualization, horizontal is not an official type for 1.3

File: dashboards-observability/public/components/explorer/visualizations/config_panel/config_editor/config_controls/config_thresholds.tsx (hunk 1 above)
Comment (mengweieric) [reply dashboards-observability/public/components/explorer/visualizations/config_panel/config_editor/config_controls/config_thresholds.tsx:34]: this relates to an issue where i found unique id was same for previously saved threshold vs newly added threshold, in that case, changing the newly added threshold would change the saved one also since they both have same th.id

File: dashboards-observability/public/components/explorer/visualizations/config_panel/config_editor/config_controls/config_thresholds.tsx (hunk 2 above)
Comment (mengweieric) [reply dashboards-observability/public/components/explorer/visualizations/config_panel/config_editor/config_controls/config_thresholds.tsx:106]: No look change, just add helper text, but having one side effect is all the input elements get lifted up a little bit, but wrapping them all achieved no look change

**Review by ps48 — APPROVED** [review]
